### DISCUSSION TITLE
Problems in creating a producer

### DISCUSSION THREAD

**Discussion by zhengcongyi** [apache/pulsar#23623] [Q&A] [answered]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

3.0.6

### Minimal reproduce step

[persistent://public/tq_business/mp_game_user_material_log-partition-0] [pulsar-log-275-486] Failed to create producer: {"errorMsg":"org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://public/tq_business/mp_game_user_material_log-partition-0) not served by this instance:10.63.103.14:8080. Please redo the lookup. Request is denied: namespace=public/tq_business","reqId":3433503011092698964,

### What did you expect to see?

[persistent://public/tq_business/mp_game_user_material_log-partition-0] [pulsar-log-275-486] Failed to create producer: {"errorMsg":"org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://public/tq_business/mp_game_user_material_log-partition-0) not served by this instance:10.63.103.14:8080. Please redo the lookup. Request is denied: namespace=public/tq_business","reqId":3433503011092698964,

### What did you see instead?

[persistent://public/tq_business/mp_game_user_material_log-partition-0] [pulsar-log-275-486] Failed to create producer: {"errorMsg":"org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://public/tq_business/mp_game_user_material_log-partition-0) not served by this instance:10.63.103.14:8080. Please redo the lookup. Request is denied: namespace=public/tq_business","reqId":3433503011092698964,

### Anything else?

[persistent://public/tq_business/mp_game_user_material_log-partition-0] [pulsar-log-275-486] Failed to create producer: {"errorMsg":"org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://public/tq_business/mp_game_user_material_log-partition-0) not served by this instance:10.63.103.14:8080. Please redo the lookup. Request is denied: namespace=public/tq_business","reqId":3433503011092698964,

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by lhotari** [comment]
@zhengcongyi Please add textual description if you file issues. If you face a problem and cannot describe it, ask a question in GitHub Discussions instead of creating issues.
I transferred this to GitHub Discussions Q&A.

**Comment by deadcoden** [comment]
I am facing the same issue(intermittently) on 3.0.4 version. Any update would be appreciated.

**Reply by lhotari** [reply]
> I am facing the same issue(intermittently) on 3.0.4 version. Any update would be appreciated.

@deadcoden Please describe the issue with steps to reproduce. That is currently missing. Have you tested on most recent version 3.0.7 (both broker & clients)?

**Reply by lhotari** [reply, marked as answer]
This problem is most likely fixed with https://github.com/apache/pulsar/pull/23515 . Please upgrade to the latest supported version release of Pulsar. Currently Pulsar 3.0.9 or Pulsar 4.0.2.

**Reply by deadcoden** [reply]
Thanks @lhotari, we will plan for an upgrade then.

**Comment by zhengcongyi** [comment]
I am facing the same issue(intermittently) on 3.0.6 version

**Reply by lhotari** [reply]
Please upgrade to get the fix, https://github.com/apache/pulsar/discussions/23623#discussioncomment-11905285 .

**Reply by zhengcongyi** [reply]
https://github.com/apache/pulsar/pull/23515  ,I have added this patch to 3.0.6, but this error still occurs。Error message:Failed to create producer: Namespace bundle for topic (persistent://low_tenant/mp_ns/xxxx-partition-2) not served by this instance:[xxxxx:8080](http://xxxxx:8080/). Please redo the lookup. Request is denied: namespace=low_tenant/mp_ns

**Reply by lhotari** [reply]
@zhengcongyi There are more fixes in 3.0.10/4.0.3 in this area.  https://github.com/apache/pulsar/pull/23884 https://github.com/apache/pulsar/pull/23984 https://github.com/apache/pulsar/pull/23988
What `loadManagerClassName` do you use?
You might need to clean up manually Zookeeper state since due to a bug, explained in https://github.com/apache/pulsar/issues/23889 and fixed by #23984 & #23988, znodes were persistent when they shouldn't have been. `bin/pulsar zookeeper-shell -server <zk-address>` to start the shell
zk shell reference: https://zookeeper.apache.org/doc/r3.9.3/zookeeperCLI.html
